### PR TITLE
[project-base] kubernetes uses relay setting for sending mails

### LINKED ISSUE #537: Mails not delivered on CI with Kubernetes
### What is happening
When CI is set up using Kubernetes according to [our documentation](https://github.com/shopsys/shopsys/blob/v7.0.0-beta2/docs/kubernetes/continuous-integration-using-kubernetes.md) and [manifests](https://github.com/shopsys/shopsys/tree/v7.0.0-beta2/project-base/kubernetes), the deployed application is not able to send e-mails (eg. when a customer registers).

Tested with whitelisted e-mail petr.heinz@shopsys.com on our internal Heimdall CI:

http://master.heimdall.netdevelo:7980/
uses Kubernetes, mail not delivered :x:

http://master-no-kubernetes.heimdall.netdevelo/
uses previous [Docker Compose solution](https://github.com/shopsys/shopsys/blob/v7.0.0-beta2/docs/cookbook/jenkins-configuration.md), mail delivered :heavy_check_mark:

### Expected result

Mails should be delivered just fine when the application is deployed in Kubernetes by CI.

### PR DESCRIPTION
**Pull request by boris-brtan**
| Q             | A
| ------------- | ---
|Description, reason for the PR| k8s deploy doesn't want to send mails via smtp container because of relay permissions and sending mail for shop is very important thing
|New feature| No 
|BC breaks| No 
|Fixes issues| resolves #537 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by boris-brtan** [comment]
please consider some article into troubleshooting or ......

File: project-base/kubernetes/deployments/smtp-server.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ spec:
             containers:
             -   name: smtp-server
                 image: namshi/smtp:latest
+                env:
+                -   name: RELAY_NETWORKS
+                    value: :192.168.0.0/16:10.0.0.0/8:172.16.0.0/12
```
Comment (grossmannmartin) [inline project-base/kubernetes/deployments/smtp-server.yml]: What exactly do IP addresses mean? Are they somehow connected to our infrastructure or they are independent?

Comment (boris-brtan) [reply project-base/kubernetes/deployments/smtp-server.yml]: allowing access from private networks that can docker containers run on

File: docs/kubernetes/how-to-set-smtp-server-container.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# How to set SMTP Server Container
+
+For sending e-mails from our application we use [SMTP server container](https://hub.docker.com/r/namshi/smtp/) in separate `k8s POD` so we need to set some additional settings as permission to be able to send e-mails from `webserver-php-fpm` POD.
+For this purpose we have set `RELAY_NETWORKS` ENV variable with all private network that can be created by docker into `/project-base/kubernetes/deployments/smtp-server.yml` for POD of smtp container ([#777](https://github.com/shopsys/shopsys/pull/777))
```
Comment (grossmannmartin) [inline docs/kubernetes/how-to-set-smtp-server-container.md]:
```suggestion
For this purpose, we have set `RELAY_NETWORKS` ENV variable with all private networks that can be created by docker into `/project-base/kubernetes/deployments/smtp-server.yml` for POD of smtp container ([#777](https://github.com/shopsys/shopsys/pull/777))  
```

**Comment by simara-svatopluk** [comment]
@boris-brtan @grossmannmartin Hi colleagues, we don't use prefixes in commit messages like `[project-base]` please don't use them next time (but do not fix it in this case as the PR is already merged to master)

**Pull request merged**